### DISCUSSION TITLE
1.0 to 2.0 upgrade guide?

### DISCUSSION THREAD

**Discussion by jpage-godaddy** [artilleryio/artillery#1993] [Artillery]
Is the 1.0 to 2.0 (still prerelease?) change breaking? If so, what are the breaking changes?

**Comment by hassy** [comment]
Hi @jpage-godaddy 👋 v2 has been the stable version for a while now. It looks like the v1 -> v2 migration guide was not carried over in a recent update to the docs we made, but I'll re-publish it shortly. The breaking changes are mostly constrained to:

- A different output format of the text log (i.e. console output) and the JSON report. If you have any code that parses either of those, that'll break
- The underlying HTTP library is now Got.js rather than Request.js. The interface of the HTTP engine remains exactly the same, but if you had any custom code that relied on a specific behavior from Request.js, that may break.

Other than that, most existing custom plugins, engines, and other code should just work.
